lib/board.rb

```diff
@@ -0,0 +1,52 @@
+class Board
+
+  attr_accessor :cells
+
+  def initialize
+    @cells = Array.new(9, " ")
+  end
+
+  def reset!
+    @cells.clear
+    @cells = Array.new(9, " ")
+  end
+
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+  end
+
+  def position(input)
+    @cells[input.to_i-1]
+  end
+
+  def full?
+    @cells.all?{|i| i == "O" || i == "X"}
+  end
+
+  def turn_count
+    @cells.count{|i| i == "X" || i == "O"}
+  end
+
+  def taken?(input)
+    if @cells[input.to_i-1] == "X" || @cells[input.to_i-1] == "O"
+      true
+    else @cells[input.to_i-1] == "" || @cells[input.to_i-1] == " "
+      false
+    end
+  end
+
+  def valid_move?(input)
+    if taken?(input.to_i) == false && input.to_i.between?(1,9) == true

```

Comment:
Avoid equality comparisons with boolean literals, relying instead on the implicit return value of the boolean expression. You can use the bang (`!`) operator to reverse or negate the truthyness of the expression.

`taken?(input.to_i) == false` is equivalent to `!taken?(input.to_i)`